pkg/storage/s3/storage.go

```diff
@@ -143,8 +149,16 @@ func (is *ObjectStorage) initRepo(name string) error {
 
 // InitRepo creates an image repository under this store.
 func (is *ObjectStorage) InitRepo(name string) error {
+	start := time.Now()
```

Comment:
why not move this into Lock()/Unlock() funcs itelf? Else all this code has to be aware of metrics!

Reply:
The reason of not moving all the logic inside Lock/Unlock functions was the latency computation logic: we need to save somewhere the starting time (could be a global variable but that is not thread safe as we calculate the starting time before lock acquiring and there is nothing between Lock/Unlock operations so other goroutines can rewrite the global starting time) and then correlate later that specific starting time with the current goroutine that processes Unlock in a defer.
Alternatively we can add this info into a context. Did you prefer the context approach instead of the PR changes?

Reply:
var start time.Time // local var so doesn't need a lock

is.Lock(&start) => func Lock() { *start := time.Now() .... }

is.Unlock(start) => func Unlock() { ... end := time.Now(), latency = end.Sub(start), metrics)

Reply:
Of course, use better varnames, "lockTimestamp", "lockStart", etc